pkg/appliance/backup/backup.go

```diff
@@ -86,6 +88,70 @@ func (b *Backup) Download(ctx context.Context, applianceID, backupID, destinatio
 	return out, nil
 }
 
+// Download a completed appliance backup with the given ID of an Appliance
+func (b *Backup) Download(ctx context.Context, applianceID, backupID, destination string) (*os.File, error) {
+	out, err := os.Create(destination)
+	if err != nil {
+		return nil, err
+	}
+	defer out.Close()
+
+	client := b.HTTPClient
+	cfg := b.APIClient.GetConfig()
+	url, err := cfg.ServerURLWithContext(ctx, "ApplianceBackupApiService.AppliancesIdBackupBackupIdGet")
+	if err != nil {
+		return nil, err
+	}
+	url = fmt.Sprintf("%s/appliances/%s/backup/%s", url, applianceID, backupID)
+	ctx = context.WithValue(ctx, api.ContextAcceptValue, fmt.Sprintf("application/vnd.appgate.peer-v%d+gpg", b.Version))
+
+	headReq, err := http.NewRequestWithContext(ctx, http.MethodHead, url, http.NoBody)
+	if err != nil {
+		return nil, err
+	}
+	head, err := client.Do(headReq)
+	if err != nil {
+		return nil, err
+	}
+
+	size := head.ContentLength
+	if err := retryDownload(ctx, client, out, url, size); err != nil {
+		return nil, err
+	}
+
+	return out, nil
+}
+
+func retryDownload(ctx context.Context, client *http.Client, f *os.File, url string, size int64) error {
+	start := int64(0)
+	buf := make([]byte, size)
+	return backoff.Retry(func() error {
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, http.NoBody)
+		if err != nil {
+			return backoff.Permanent(err)
+		}
+		req.Header.Set("Range", fmt.Sprintf("bytes=%d-", start))
+
+		res, err := client.Do(req)
```

Comment:
Personally, I would like to see DEBUG logging here to show what range we are currently fetching, in case we have to troubleshoot issues in the future.

Reply:
I can add a log here, but running with the `--debug` flag will log every request/response anyway. It's part of the client we use.

Reply:
`--debug` shows the request/response but it doesn't display the range headers

```
2024/01/25 07:33:41
GET /admin/appliances/ec806b3b-0385-4402-afee-12928087806b/backup/f827bfce-bb7d-11ee-82db-005056b51bc5/status HTTP/1.1
Host: envy-10-98-136-2.devops:8443
User-Agent: sdpctl/0.0.0-dev
Accept: application/vnd.appgate.peer-v20+json
Authorization: Bearer REDACTED
Accept-Encoding: gzip
```

[sdpctl-appliance-backup.log](https://github.com/appgate/sdpctl/files/14051681/sdpctl-appliance-backup.log)